lib/client-assets.php

```diff
@@ -187,12 +187,13 @@ function gutenberg_register_packages_scripts() {
 		// For example, `…/build/a11y/index.js` becomes `wp-a11y`.
 		$handle = 'wp-' . basename( dirname( $path ) );
 
-		// Replace `.js` extension with `.deps.json` to find the generated dependencies file.
-		$dependencies_file = substr( $path, 0, -3 ) . '.deps.json';
-
-		$dependencies = is_readable( $dependencies_file )
-			? json_decode( file_get_contents( $dependencies_file ) )
-			: array();
+		// Replace `.js` extension with `.asset.php` to find the generated dependencies file.
+		$asset_file   = substr( $path, 0, -3 ) . '.asset.php';
+		$asset        = file_exists( $asset_file )
+			? require_once( $asset_file )
```

Comment:
Is the `require_once()` important? That is, why not use `require()`?

With `require_once()`, if `wp_default_scripts` runs more than once, the second (any any subsequent) run of each `require_once()` will just return `true`, and we'll lose all the dependencies for the scripts.

Reply:
In theory, this function doesn't run more than once. However, `require` makes more sense because the file returns a static value rather than defining a function or class.

Reply:
https://github.com/WordPress/gutenberg/issues/18596